benchmarks/benchmark.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <iostream>
+#include <chrono>
+#include <ctime>
+#include <cmath>
+#include <string>
+#include <vector>
+
+class benchmark
+{
+  using clock = std::chrono::system_clock;
+public:
+  benchmark(std::string const &name) : name_(name) {}
+  void print_header()
+  {
+    std::cout << "# benchmark : " << name_ << '\n'
+              << "# ..." << std::endl;
+  }
+  virtual void setup(long) {}
+  virtual void operation(long) {}
+  virtual void teardown() {}
+  
+  void run(std::vector<long> const &sizes, unsigned times = 1)
```

Comment:
Agreed on `times`. As to the single `sizes` parameter: I'm merely trying to provide a unique way to iterate over *a* "size" parameter to allow to sweep across a range of "sizes". What "size" means for a specific benchmark is still up to the implementor. It could mean the size of a square matrix, or it could be something else.
Given the abstract nature of benchmarks, I don't want to impose any specific meaning. But neither does it seem appropriate to have some benchmarks use a single "size" parameter, and some benchmarks two or even more.